lib/s3routes/routesUtils.ts

```diff
@@ -408,15 +420,18 @@ function retrieveData(
         client, implName, config, kms, metadata, locStorageCheckFn, vault);
     return eachSeries(locations,
         (current, next) => data.get(current, response, log,
-            (err: any, readable: http.IncomingMessage) => {
+            (err: Error, readable: http.IncomingMessage) => {
                 const cbOnce = jsutil.once(next);
                 // NB: readable is of IncomingMessage type
                 if (err) {
-                    log.error('failed to get object', {
+                    // In this case, we do not destroy the response:
+                    // nothing was sent to the client yet, so we
+                    // can return a proper error.
+                    log.debug('failed to get object', {
                         error: err,
                         method: 'retrieveData',
                     });
-                    _destroyResponse();
+                    _destroyReadable(readable);
```

Comment:
we used to destroy the `response`, now we destroy the `request` : can't we just "not destroy anything", and just let the readable be destroyed normally? (the readable will be closed once the response is closed anyway)
or is there a reason/need for this explicit "early" destroy?